app/src/main/java/com/awscherb/cardkeeper/data/handler/BaseRealmHandler.java

```diff
@@ -84,16 +67,67 @@
     //================================================================================
 
     @Override
-    public Observable<Void> deleteObject(T object) {
+    public Completable deleteObject(T object) {
         Realm realm = Realm.getDefaultInstance();
```

Comment:
You are obtaining Realm isntances at different points. Once you start attaching non main thread schedulers it will cause problems.